packages/reports/src/Stores/Modules/Statement/__tests__/statement-store.spec.js

```diff
@@ -0,0 +1,218 @@
+        statement_store.has_loaded_all = true;
+        statement_store.is_loading = true;
+
+        statement_store.clearTable();
+
+        expect(statement_store.data).toEqual([]);
+        expect(statement_store.has_loaded_all).toBe(false);
+        expect(statement_store.is_loading).toBe(false);
+    });
+
+    it('should clear date filters', () => {
+        statement_store.clearDateFilter();
+
+        expect(statement_store.date_from).toBeNull();
+        expect(statement_store.date_to).toBe(toMoment().startOf('day').add(1, 'd').subtract(1, 's').unix());
+        expect(statement_store.partial_fetch_time).toBe(0);
+    });
+
+    it('should return false when shouldFetchNextBatch is called', () => {
+        statement_store.has_loaded_all = true;
+        statement_store.is_loading = true;
+        expect(statement_store.shouldFetchNextBatch(false)).toBe(false);
+    });
+
+    it('should return true when shouldFetchNextBatch is called', () => {
+        statement_store.has_loaded_all = false;
+        statement_store.is_loading = false;
+        statement_store.partial_fetch_time = '2024-07-08 07:10:00 GMT';
+        expect(statement_store.shouldFetchNextBatch(false)).toBe(true);
+    });
+
+    it('should fetch next batch correctly', async () => {
+        WS.statement.mockResolvedValue(response);
+        statement_store.shouldFetchNextBatch = jest.fn().mockReturnValue(true);
+
+        await statement_store.fetchNextBatch();
+
+        expect(statement_store.is_loading).toBe(false);
+        expect(WS.statement).toHaveBeenCalled();
+        expect(statement_store.data.length).toBe(2);
+    });
+
+    it('should handle date change when both from and to date values are provided', () => {
+        const date_values = {
+            from: '2021-01-01',
+            to: '2021-01-31',
+            is_batch: false,
+        };
+
+        statement_store.handleDateChange(date_values);
+
+        expect(statement_store.date_from).toBe(toMoment(date_values.from).unix());
+        expect(statement_store.date_to).toBe(toMoment(date_values.to).unix());
+    });
+
+    it('should handle date change when is_batch value is true', () => {
+        const date_values = {
+            from: '',
+            to: '2021-01-31',
+            is_batch: true,
+        };
+
+        statement_store.handleDateChange(date_values);
+
+        expect(statement_store.date_from).toBe(null);
+        expect(statement_store.date_to).toBe(toMoment(date_values.to).unix());
+    });
+
+    it('should handle filter change correctly', () => {
+        const filterValue = 'deposit';
+        WS.statement.mockResolvedValue(response);
+        statement_store.shouldFetchNextBatch = jest.fn().mockReturnValue(true);
+
+        statement_store.handleFilterChange(filterValue);
+
+        expect(statement_store.action_type).toBe(filterValue);
+    });
+
+    it('should call fetchNextBatch when fetchOnScroll is called with left less than 1500', () => {
```

Comment:
since we are checking for the result of fetchNextBatch() call instead of its own call, perhaps it's better to rename it:
```suggestion
    it('should fetch statement when fetchOnScroll is called with left less than 1500', () => {
```
@vinu-deriv  what do you think? :)